src/operator/aggregate/average.rs

```diff
@@ -153,11 +154,58 @@ where
         let aggregate =
             self.aggregate_linear::<TS, _, _>(move |key, val| Avg::new(f(key, val), 1isize));
 
-        // TODO: We can probably use some sort of `.map_index_owned()` here since in all
-        // likelihood we'll be the only consumer of the aggregated value
-        // (meaning we wouldn't need to clone the average's sum and would be
-        // able to elide one clone of the key value)
-        let average = aggregate.map_index(|(k, avg)| (k.clone(), (avg.sum.clone()) / avg.count));
+        // We're the only possible consumer of the aggregated stream so we can use an
+        // owned consumer to skip any cloning
+        let average = aggregate.apply_owned_named("MapAverage", |aggregate| {
```

Comment:
Any chance you could add a test (proptest?) for the `average` operator?

Reply:
Also, how does this affect the performance of q4, which is the only benchmark that uses `average` at the moment?

Reply:
On my machine with 1 thread & 1 generator I get 89.412s for the previous impl and 74.469s with my change